### PR TITLE
Download files with `file-saver`

### PR DESCRIPTION
**Pull request by tsauerwein**
Use the library [file-saver](https://github.com/eligrey/FileSaver.js) to download KML/GPX/CSV files.

Currently I can't test properly, but file-saver has support for all modern browsers.

Closes https://github.com/camptocamp/ngeo/issues/1776

### CODE REVIEW

File: src/services/featurehelper.js [hunk 1]
```diff
@@ -88,6 +89,13 @@ ngeo.FeatureHelper = function($injector, $filter) {
    */
   this.projection_;
 
+  /**
+   * Download service.
+   * @type {ngeo.Download}
+   * @private
+   */
+  this.download_ = $injector.get('ngeoDownload');
```
Comment (gberaudo) [inline src/services/featurehelper.js]: Shouldn't you use Angular automatic injection instead of doing it manually here?

Comment (tsauerwein) [inline src/services/featurehelper.js]: There isn't a clear guideline in ngeo when to use `$injector` and when to use automatic injection.

Comment (sbrunner) [inline src/services/featurehelper.js]: For me the injector should be used when the dependency is optional (and in the gmf controller for simplification).
If it's not like that we can clarify it later...

**Comment by fredj** [comment]
We can merge this PR and create a new issue for the injector usage

**Comment by sbrunner** [comment]
@tsauerwein up :-)

**Comment by fredj** [comment]
@tsauerwein please rebase and merge

File: src/services/featurehelper.js (hunk 1 above)
Comment (tsauerwein) [inline src/services/featurehelper.js]: I opened an  issue: https://github.com/camptocamp/ngeo/issues/1865

**Comment by tsauerwein** [comment]
Tested with BrowserStack on Windows 10 and Edge 14, merging.

**Pull request merged**